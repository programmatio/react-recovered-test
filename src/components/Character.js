import React from "react";
import PropTypes from "prop-types";

const Character = ({ character }) => {
  return (
    <dl>
      <dt>Name</dt>
      <dd>{character.name}</dd>
      <dt>Heigh</dt>
      <dd>{character.height}</dd>
      <dt>Hair Color</dt>
      <dd>{character.hair_color}</dd>
      <dt>Eye Color</dt>
      <dd>{character.eye_color}</dd>
    </dl>
  );
};
Character.propTypes = {
  character: PropTypes.object.isRequired
};

export default Character;
