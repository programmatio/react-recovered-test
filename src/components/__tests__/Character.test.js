import React from "react";
import { shallow } from "enzyme";
import Character from "../Character";

const props = {
  character: {
    name: "test",
    height: "test",
    hair_color: "test",
    eye_color: "test"
  }
};
const wrapper = shallow(<Character {...props} />);

describe("<Character />", () => {
  it("should match the snapshot", function() {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render definition list", function() {
    expect(wrapper.find("dl").length).toBe(1);
  });
});
