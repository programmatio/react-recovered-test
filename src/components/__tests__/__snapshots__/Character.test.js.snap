// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Character /> should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Character
    character={
        Object {
            "eye_color": "test",
            "hair_color": "test",
            "height": "test",
            "name": "test",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <dt>
          Name
</dt>,
        <dd>
          test
</dd>,
        <dt>
          Heigh
</dt>,
        <dd>
          test
</dd>,
        <dt>
          Hair Color
</dt>,
        <dd>
          test
</dd>,
        <dt>
          Eye Color
</dt>,
        <dd>
          test
</dd>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Name",
        },
        "ref": null,
        "rendered": "Name",
        "type": "dt",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "dd",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Heigh",
        },
        "ref": null,
        "rendered": "Heigh",
        "type": "dt",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "dd",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Hair Color",
        },
        "ref": null,
        "rendered": "Hair Color",
        "type": "dt",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "dd",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Eye Color",
        },
        "ref": null,
        "rendered": "Eye Color",
        "type": "dt",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "dd",
      },
    ],
    "type": "dl",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <dt>
            Name
</dt>,
          <dd>
            test
</dd>,
          <dt>
            Heigh
</dt>,
          <dd>
            test
</dd>,
          <dt>
            Hair Color
</dt>,
          <dd>
            test
</dd>,
          <dt>
            Eye Color
</dt>,
          <dd>
            test
</dd>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Name",
          },
          "ref": null,
          "rendered": "Name",
          "type": "dt",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "dd",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Heigh",
          },
          "ref": null,
          "rendered": "Heigh",
          "type": "dt",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "dd",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Hair Color",
          },
          "ref": null,
          "rendered": "Hair Color",
          "type": "dt",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "dd",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Eye Color",
          },
          "ref": null,
          "rendered": "Eye Color",
          "type": "dt",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "dd",
        },
      ],
      "type": "dl",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
